app/api/role_invites.py

```diff
@@ -112,8 +112,10 @@ def before_update_object(self, role_invite, data, view_kwargs):
         """
         user = User.query.filter_by(email=role_invite.email).first()
         if user:
-            if not has_access('is_user_itself', user_id=user.id):
-                raise UnprocessableEntity({'source': ''}, "Only users can edit their own status")
+            if not has_access('is_organizer', event_id=role_invite.event_id) and not has_access('is_user_itself',
```

Comment:
@shreyanshdwivedi Shouldn't `Owner` be also there?

Reply:
@kushthedude owner has access as an organizer too - 
`app/api/helpers/permission_manager.py`
```py
if user.is_owner(kwargs['event_id']) or user.is_organizer(kwargs['event_id']):
        return view(*view_args, **view_kwargs)
```